### DISCUSSION TITLE
Cache::block() did not  work

### DISCUSSION THREAD

**Discussion by gakkiismywife** [laravel/framework#38247] [Q&A]
```
$lock = Cache::lock('withdraw:' . $record->uuid, 10, rand());
try {
            $lock->block(5);
} catch (LockTimeoutException $exception) {
} catch (Exception $exception) {
    $lock->release();
}
```
This is my code
My cache driver is redis and the redis was working with  master and slave mode
When I use Jmeter to Send 3 requests at one second  and they are both success
So the requests were not blocked

**Comment by sayabakso** [comment]
I'm having the same issue, I use Jmeter as well to send a request to dispatch 2 concurrent job to test race condition with redis atomic lock
After logging the result of redis setnx, both jobs got return value of 1
Apparently, the redis setnx cannot prevent race condition because the function itself can get into race condition situation
Maybe real world cases won't have 2 processes call redis setnx function at the same time, but knowing with Jmeter is still doesn't prevent race condition 100% make it a bit unreliable, but is this how it's supposed to be?
